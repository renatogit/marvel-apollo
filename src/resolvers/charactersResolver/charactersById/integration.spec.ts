const CharactersResolver = require('.');
const DataSourcesCharactersAPI = require('@/dataSources/characters');
const ConfigServer = require('@/configs/test');
const QUERY_GET_CHARACTERS_BY_ID = require('@query/characters/queryCharactersById');
const MOCK_GET_CHARACTERS_BY_ID = require('@mock/characters/mockCharacters');

describe('resolvers/charactersResolver/integration', () => {
	it('fetches characters and generate snapshots', async () => {
		const charactersId = {charactersId: 'fake-code'};

		const args = {
			dataSources: DataSourcesCharactersAPI,
			resolver: CharactersResolver.Query.charactersById,
			mock: MOCK_GET_CHARACTERS_BY_ID,
			variables: {charactersId},
			entity: 'characters',
			typeQuery: 'charactersById',
		};

		const {server} = await ConfigServer(args);

		const {body} = await server.executeOperation({
			query: QUERY_GET_CHARACTERS_BY_ID,
			variables: charactersId,
		});

		expect(body).toMatchSnapshot();
	});
});
