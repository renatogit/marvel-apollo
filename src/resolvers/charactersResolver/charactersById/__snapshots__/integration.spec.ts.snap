// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/charactersResolver/integration fetches characters and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "charactersById": {
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 20,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "comics": null,
              "description": "mock data",
              "events": null,
              "id": 1011334,
              "modified": "mock data",
              "name": "mock data",
              "series": null,
              "stories": null,
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 1564,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
